app/shared/inputs/SignupForm.js

```diff
@@ -0,0 +1,174 @@
+    }
+  };
+  return (
+    showForm && (
+      <form
+        noValidate
+        onSubmit={(e) => e.preventDefault()}
+        id={labelId}
+        className="flex w-full"
+      >
+        <>
+          <div
+            className={`flex flex-col w-full items-center justify-center lg:flex-row mb-10`}
+          >
+            <div className="flex flex-col w-full items-center lg:items-start">
+              <div className="flex flex-row w-full md:max-w-[300px] items-start mr-5">
+                <span className="text-sm ml-5">First Name</span>
+                <span className="text-sm text-red-600 ml-1">*</span>
+              </div>
+              <TextField
+                name="firstName"
+                onChange={(e) => {
+                  setFirstName(e.target.value);
+                }}
+                fullWidth
+                value={firstName}
+                placeholder="Jane"
+                className="py-4 pl-4 mb-5 rounded-lg w-full bg-slate-50 border-[1px] border-indigo-200 text-black  max-w-full md:max-w-[300px]"
+              />
+            </div>
+
+            <div className="flex flex-col w-full items-center lg:items-start mt-5 lg:mt-0 lg:ml-5">
+              <div className="flex flex-row w-full md:max-w-[300px] items-start mr-5">
+                <span className="text-sm ml-5">Last Name</span>
+                <span className="text-sm text-red-600 ml-1">*</span>
+              </div>
+              <TextField
+                name="lastName"
+                size="medium"
+                fullWidth
+                onChange={(e) => {
+                  setLastName(e.target.value);
+                }}
+                value={lastName}
+                placeholder="Doe"
+                className="py-4 pl-4 mb-5 rounded-lg w-full bg-slate-50 border-[1px] border-indigo-200 text-black max-w-full md:max-w-[300px]"
+              />
+            </div>
+
+            <div className="flex flex-col w-full items-center lg:items-start mt-5 lg:mt-0 lg:ml-5">
+              <div className="flex flex-row w-full md:max-w-[300px] items-start mr-5">
+                <span className="text-sm ml-5">Email</span>
+                <span className="text-sm text-red-600 ml-1">*</span>
+              </div>
+              <EmailInput
+                name="email"
+                onChangeCallback={(e) => {
+                  setEmail(e.target.value);
+                }}
+                value={email}
+                placeholder="jane.doe@email.com"
+                useLabel={false}
+                className="py-4 pl-4 mb-5 rounded-lg w-full bg-slate-50 border-[1px] border-indigo-200 text-black max-w-full md:max-w-[300px]"
+              />
+            </div>
+
+            <div className="flex flex-col w-full items-center lg:items-start mt-5 lg:mt-0 lg:ml-5">
+              <div className="flex flex-row w-full md:max-w-[300px] items-start mr-5">
+                <span className="text-sm ml-5">Phone Number (optional)</span>
```

Comment:
you notice how you repeat your code 4 times?
add the elements in an array and iterate over it.